scalafix-core/shared/src/main/scala/scalafix/internal/config/DisableConfig.scala

```diff
@@ -1,48 +1,111 @@
+    safeBlock: Symbol.Global,
+    @Description(
+      "The unsafe symbols that are banned unless inside a 'safe' block")
+    symbols: List[CustomMessage[Symbol.Global]])
+
+object UnlessInsideBlock {
+  implicit val surface: Surface[UnlessInsideBlock] =
+    generic.deriveSurface[UnlessInsideBlock]
+  // NOTE(olafur): metaconfig.generic.deriveDecoder requires a default base values.
+  // Here we require all fields to be provided by the user so we write the decoder manually.
+  implicit val reader: ConfDecoder[UnlessInsideBlock] =
+    ConfDecoder.instanceF[UnlessInsideBlock] {
+      case c: Conf.Obj =>
+        (c.get[Symbol.Global]("safeBlock") |@|
+          c.get[List[CustomMessage[Symbol.Global]]]("symbols")).map {
+          case (a, b) => UnlessInsideBlock(a, b)
+        }
+      case _ => Configured.NotOk(ConfError.message("Wrong config format"))
+    }
+}
+
 case class DisableConfig(
-    @Description("The list of symbols to disable.")
-    @ExampleValue(
-      """[
-        |  # With custom message (recommended)
-        |  {
-        |    symbol = "scala.Predef.any2stringadd"
-        |    message = "Use explicit toString before calling +"
-        |  }
-        |  {
-        |    symbol = "scala.Any"
-        |    message = "Explicitly type annotate Any if this is intentional"
-        |  }
-        |  # Without custom message (discouraged)
-        |  "com.Lib.implicitConversion"
-        |]""".stripMargin)
-    symbols: List[CustomMessage[Symbol.Global]] = Nil
+    @Description("The list of symbols to disable only in the actual sources.")
+    @ExampleValue("""
+                    |[
+                    |  {
+                    |    symbol = "scala.Any.asInstanceOf"
+                    |    message = "use pattern-matching instead"
+                    |  }
+                    |]""".stripMargin)
+    symbols: List[CustomMessage[Symbol.Global]] = Nil,
+    @Description(
+      "The list of symbols to disable, also blocks symbols in the generated code")
+    @ExampleValue("""
+                    |[
+                    |  {
+                    |    symbol = "scala.Predef.any2stringadd"
+                    |    message = "use explicit toString be calling +"
+                    |  }
+                    |]""".stripMargin)
+    ifSynthetic: List[CustomMessage[Symbol.Global]] = Nil,
+    @Description(
+      "The list of symbols to disable unless they are in the given block.")
+    @ExampleValue("""
+                    |[
+                    |  {
+                    |    safeBlock = "scala.util.Try"
+                    |    symbols = [
+                    |      {
+                    |        symbol = "scala.Option.get"
+                    |        message = "the function may throw an exception"
+                    |      }
+                    |    ]
+                    |  }
+                    |]
+      """.stripMargin)
+    unlessInsideBlock: List[UnlessInsideBlock] = Nil
```

Comment:
What do you think about renaming this to "unlessInsideSafeBlock"? I think it makes sense to be consistent betweeen `.unlessInsideBlock` and `safeBlock =`. Standalone "Block" is too generic so I think "safeBlock" may be clearer, but I'll leave it to you and @fommil to decide what makes most sense.

Reply:
the concept of "safe"ness is perhaps limited to my usecase. I can imagine people wanting to use this for other purposes, such as logging or monitoring, in which case "safe" will be a misnomer.

Reply:
Fair enough. What should the "safe symbol" be called then?

`Disable.unlessInside = [ ??? ]`